### PR TITLE
[MINOR]Simplify boolean expression

### PR DESCRIPTION
**Pull request by 10110346**
## What changes were proposed in this pull request?

Comparing whether Boolean expression is equal to true is redundant
For example:
The datatype of `a` is boolean.
Before:
if (a == true)
After:
if (a)

## How was this patch tested?
N/A

### CODE REVIEW

**Comment by maropu** [comment]
How did you check this? I run`grep -e "== true" -e "== false" -e "!= true" -e "!= false"` in all the source code, and I quickly looked over the result and I found at least one more file (`SecurityManagerSuite`) having this comparison, too.

**Comment by 10110346** [comment]
> 
> 
> How did you check this? I run`grep -e "== true" -e "== false" -e "!= true" -e "!= false"` in all the source code, and I quickly looked over the result and I found at least one more file (`SecurityManagerSuite`) having this comparison, too.

I checked it using `IntelliJ IDEA---Analyze`.
It  seems that `===` has not been detected

**Comment by dilipbiswal** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala [hunk 1]
```diff
@@ -772,7 +772,7 @@ private[deploy] object Worker extends Logging {
     // more detail see SPARK-20989.
     val externalShuffleServiceEnabled = conf.get(config.SHUFFLE_SERVICE_ENABLED)
     val sparkWorkerInstances = scala.sys.env.getOrElse("SPARK_WORKER_INSTANCES", "1").toInt
-    require(externalShuffleServiceEnabled == false || sparkWorkerInstances <= 1,
+    require(!externalShuffleServiceEnabled || sparkWorkerInstances <= 1,
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala]: I agree with removing `== true`. There is some minor argument against `!` vs `== false` as the `!` can be easier to overlook. Although it comes first, I suppose you could say. I'm neutral on those.

File: core/src/test/scala/org/apache/spark/util/BoundedPriorityQueueSuite.scala [hunk 2]
```diff
@@ -29,12 +29,12 @@ class BoundedPriorityQueueSuite extends SparkFunSuite {
     pq += 0.3
     pq += 0.01
 
-    assert(pq.isEmpty == false)
+    assert(!pq.isEmpty)
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/util/BoundedPriorityQueueSuite.scala]: How about `.nonEmpty` instead

**Review by srowen — COMMENTED** [review]
One minor comment

File: core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala (hunk 1 above)
Comment (10110346) [reply core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala]: It sounds reasonable, thanks

File: core/src/test/scala/org/apache/spark/UnpersistSuite.scala [hunk 3]
```diff
@@ -31,7 +31,7 @@ class UnpersistSuite extends SparkFunSuite with LocalSparkContext with TimeLimit
     rdd.count
     assert(sc.persistentRdds.isEmpty === false)
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/UnpersistSuite.scala]: You're welcome to grep for `\.isEmpty ={2,3} false` and make them .`nonEmpty` just while we're here, but this looks good.

**Review by srowen — APPROVED** [review]

File: core/src/test/scala/org/apache/spark/UnpersistSuite.scala (hunk 3 above)
Comment (10110346) [reply core/src/test/scala/org/apache/spark/UnpersistSuite.scala]: Only found this one using `grep`， thanks

**Comment by viirya** [comment]
> The comparison between Boolean expression and Boolean literal is redundant.

The PR description can be improved.

**Comment by 10110346** [comment]
> 
> 
> > The comparison between Boolean expression and Boolean literal is redundant.
> 
> The PR description can be improved.

Thanks, updated.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**